android/app/src/main/java/com/expensify/chat/intentHandler/TextIntentHandler.kt

```diff
@@ -0,0 +1,82 @@
+package com.expensify.chat.intentHandler
+
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import com.expensify.chat.utils.FileUtils
+
+
+class TextIntentHandler(private val context: Context) : AbstractIntentHandler() {
+    override fun handle(intent: Intent): Boolean {
+        super.clearTemporaryFiles(context)
+        when(intent.action) {
+            Intent.ACTION_SEND -> {
+                handleTextIntent(intent, context)
+                onCompleted()
+                return true
+            }
+        }
+        return false
+    }
+
+    private fun handleTextIntent(intent: Intent, context: Context) {
+        when {
+            intent.type == "text/plain" -> {
+                val extras = intent.extras
+                if (extras != null) {
+                    when {
+                        extras.containsKey(Intent.EXTRA_STREAM) -> {
+                            handleTextFileIntent(intent, context)
+                        }
+                        extras.containsKey(Intent.EXTRA_TEXT) -> {
+                            handleTextPlainIntent(intent, context)
+                        }
+                        else -> {
+                            throw UnsupportedOperationException("Unknown text/plain content")
+                        }
+                    }
+                }
+            }
+            Regex("text/.*").matches(intent.type ?: "") -> handleTextFileIntent(intent, context)
+            else -> throw UnsupportedOperationException("Unsupported MIME type: ${intent.type}")
+        }
+    }
+
+    private fun handleTextFileIntent(intent: Intent, context: Context) {
+        (intent.getParcelableExtra<Uri>(Intent.EXTRA_STREAM))?.let { fileUri ->
+            if (fileUri == null) {
+                return
+            }
+
+            val resultingPath: String? = FileUtils.copyUriToStorage(fileUri, context)
+            if (resultingPath != null) {
+                val shareFileObject = ShareFileObject(resultingPath, intent.type)
+
+                val sharedPreferences = context.getSharedPreferences(IntentHandlerConstants.preferencesFile, Context.MODE_PRIVATE)
+                val editor = sharedPreferences.edit()
+                editor.putString(IntentHandlerConstants.shareObjectProperty, shareFileObject.toString())
+                editor.apply()
```

Comment:
I think this is duplicated in `handleTextPlanIntent`, can we extract it to a separate method?